### DISCUSSION TITLE
`messaging/internal-error`: what does it mean and how should we handle it?

### DISCUSSION THREAD

**Discussion by eliw00d** [firebase/firebase-admin-node#1618] [Q&A] [answered]
I frequently get `messaging/internal-error` errors when sending token or tokens (multicast) messages via the Node.js SDK. The message is simply `Internal error encountered.`. 

It seems to happen for any kind of device: iOS, Android, or Web.

I check for the following error codes:
- `messaging/invalid-registration-token`
- `messaging/registration-token-not-registered`
- `messaging/invalid-argument` (with message `The registration token is not a valid FCM registration token`)

but am not sure how to handle `messaging/internal-error`. Should I consider it a retry-able event, like the server being down? Or can it sometimes mean the token is invalid and we should no longer try sending to it?

Update: I used https://iid.googleapis.com/iid/info on a few recent failures to see if the tokens were bad and they were all 200 OK responses.

**Comment by lahirumaramba** [comment, marked as answer]
That should be a retry-able event. If you consistently seeing this issue for a specific token please contact Firebase support where you can share more details specific to your project: https://firebase.google.com/support/troubleshooter

cc: @chong-shao

**Reply by eliw00d** [reply]
Sounds good. I will reach out as it happens quite often. Thanks!

**Reply by eliw00d** [reply]
@lahirumaramba How much does the SDK retry for this error and how much should we retry if we experience it? It sounds like we are getting these errors not due to an outage but other circumstances.
